integration/utmp_integration_test.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/bpf"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/pam"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/regular"
+	"github.com/gravitational/teleport/lib/srv/uacc"
+	"github.com/gravitational/teleport/lib/sshutils"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+)
+
+// teleportTestUser is additional user used for tests
+const teleportTestUser = "teleport-test"
+
+// wildcardAllow is used in tests to allow access to all labels.
+var wildcardAllow = services.Labels{
+	services.Wildcard: []string{services.Wildcard},
+}
+
+type SrvCtx struct {
+	srv        *regular.Server
+	signer     ssh.Signer
+	server     *auth.TestTLSServer
+	testServer *auth.TestAuthServer
+	clock      clockwork.FakeClock
+	nodeClient *auth.Client
+	nodeID     string
+	utmpPath   string
+}
+
+// TestRootUTMPEntryExists verifies that user accounting is done on supported systems.
+func TestRootUTMPEntryExists(t *testing.T) {
+	s := newSrvCtx(t)
+	up, err := newUpack(s, teleportTestUser, []string{teleportTestUser}, wildcardAllow)
+	require.NoError(t, err)
+
+	sshConfig := &ssh.ClientConfig{
+		User:            teleportTestUser,
+		Auth:            []ssh.AuthMethod{ssh.PublicKeys(up.certSigner)},
+		HostKeyCallback: ssh.FixedHostKey(s.signer.PublicKey()),
+	}
+
+	client, err := ssh.Dial("tcp", s.srv.Addr(), sshConfig)
+	require.NoError(t, err)
+	defer func() {
+		err := client.Close()
+		require.NoError(t, err)
+	}()
+
+	se, err := client.NewSession()
+	require.NoError(t, err)
+	defer se.Close()
+
+	modes := ssh.TerminalModes{
+		ssh.ECHO:          0,     // disable echoing
+		ssh.TTY_OP_ISPEED: 14400, // input speed = 14.4kbaud
+		ssh.TTY_OP_OSPEED: 14400, // output speed = 14.4kbaud
+	}
+
+	require.NoError(t, se.RequestPty("xterm", 80, 80, modes), nil)
+	err = se.Shell()
+	require.NoError(t, err)
+	time.Sleep(5 * time.Second)
```

Comment:
What if the entry does not appear within the specified interval? Is there a way to ensure the entry appears w/o resorting to timeout?
One idea to mitigate the fixed timeout is poll the file in shorter steps.

Reply:
It doesn't really send an event but we could keep repolling for a few minutes or so.

Reply:
Now repolls for 5 minutes before giving up.